File: sections/229.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
const majorityElement = (nums) => {
    const n = nums.length / 3
    const map = {}
    const result = []

    nums.forEach((d) => {
        if (map[d] === undefined) {
            map[d] = 0
        }

        map[d]++

        if (map[d] > n && map[d] <= n + 1) {
            result.push(d)
        }
    })

    return result
}

File: sections/229.py
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        n = len(nums) / 3
        map = {}
        r = []

        for d in nums:
            if map.get(d) is None:
                map[d] = 0

            map[d] += 1

            if map[d] > n and map[d] <= n + 1:
                r.append(d)

        return r
